src/linear_algebra/basis.lean

```diff
@@ -92,45 +92,49 @@ lemma is_basis.injective [nontrivial R] (hv : is_basis R v) : injective v :=
 lemma is_basis.range (hv : is_basis R v) : is_basis R (λ x, x : range v → M) :=
 ⟨hv.1.to_subtype_range, by { convert hv.2, ext i, exact ⟨λ ⟨p, hp⟩, hp ▸ p.2, λ hi, ⟨⟨i, hi⟩, rfl⟩⟩ }⟩
 
+variable (v)
+
 /-- Given a basis, any vector can be written as a linear combination of the basis vectors. They are
 given by this linear map. This is one direction of `module_equiv_finsupp`. -/
 def is_basis.repr : M →ₗ (ι →₀ R) :=
 (hv.1.repr).comp (linear_map.id.cod_restrict _ hv.mem_span)
 
-lemma is_basis.total_repr (x) : finsupp.total ι M R v (hv.repr x) = x :=
+variable {v}
+
+lemma is_basis.total_repr (x) : finsupp.total ι M R v (hv.repr v x) = x :=
 hv.1.total_repr ⟨x, _⟩
 
-lemma is_basis.total_comp_repr : (finsupp.total ι M R v).comp hv.repr = linear_map.id :=
+lemma is_basis.total_comp_repr : (finsupp.total ι M R v).comp (hv.repr v) = linear_map.id :=
```

Comment:
It feels quite clumsy to me to have to pass `v` into `hv.repr` when `hv` already knows it is a hypothesis about `v`. Can this be fixed by instead defining a custom pretty-printer for `is_basis`?

Reply:
Note that `hv` will usually be pretty printed as `_` unless it is trivial, so this isn't as redundant as it looks.

Reply:
Anyway you don't have to pass `v` in, you can also write `_` if you want it inferred.

Reply:
Right, I understand that the motivation is to avoid underscores (although that should probably be in the commit message) - what I question is whether that motivation is sufficient to justify this change, and whether there's another way to address it.

Reply:
How hard is it to make an option that adds type annotations to underscores?